mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -411,7 +411,10 @@ private[ml] object DefaultParamsWriter {
       paramMap: Option[JValue] = None): Unit = {
     val metadataPath = new Path(path, "metadata").toString
     val metadataJson = getMetadataToSave(instance, sc, extraMetadata, paramMap)
-    sc.parallelize(Seq(metadataJson), 1).saveAsTextFile(metadataPath)
+    val spark = SparkSession.builder().sparkContext(sc).getOrCreate()
+    // Note that we should write single file. If there are more than one row
+    // it produces more partitions.
+    spark.createDataFrame(Seq(Tuple1(metadataJson))).write.text(metadataPath)
```

Comment:
It seems that the previous code did not have `repartition(1)`. 
> This PR proposes to remove repartition(1) when writing metadata in ML/MLlib. It already writes one file.

Maybe, I missed the context. Does the existing code generate multiple file, @HyukjinKwon ?
> sc.parallelize(Seq(metadataJson), 1).saveAsTextFile(metadataPath)